src/sdk/namespace_merge.js

```diff
@@ -25,9 +25,18 @@ class NamespaceMerge {
     // OBJECT LIST //
     /////////////////
 
-    list_objects(params, object_sdk) {
-        return this._ns_map(ns => ns.list_objects(params, object_sdk))
-            .then(res => this._handle_list(res, params));
+    async list_objects(params, object_sdk) {
+        let res = (await this._ns_map(async function(ns) {
+            try {
+                let ns_objects = await ns.list_objects(params, object_sdk);
+                return ns_objects;
+            } catch (err) {
+                if (err.code === 'NoSuchBucket') return;
+                throw err;
+            }
+        })).filter(objects => objects);
```

Comment:
This is a bit confusing to read. I would maybe advise to either use the isUndefined or even add a custom omit value in the return above. At least explain using a comment that even empty buckets have an object reply and you won't omit them.